docs/data-masking-function-list.md

```diff
@@ -0,0 +1,888 @@
+
+### Parameters
+
+None.
+
+### Returns
+
+A NIN string.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_uk_nin();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +----------------------+
+    | gen_rnd_uk_nin()     |
+    +----------------------+
+    | BK189806             |
+    +----------------------+
+    ```
+
+
+## gen_rnd_us_phone()
+
+
+
+Generates a United States phone number with the `555` area code.
+
+### Parameters
+
+None
+
+### Returns
+
+Returns a United States phone number.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_us_phone();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +--------------------+
+    | gen_rnd_us_phone() |
+    +--------------------+
+    | 1-555-249-2029     |
+    +--------------------+
+    ```
+
+## gen_rnd_uuid()
+
+
+
+Generates a Universally Unique Identifier (UUID).
+
+### Parameters
+
+None.
+
+### Returns
+
+Returns a UUID as a string.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_uuid();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +------------------------------------+
+    | gen_rnd_uuid()                     |
+    +------------------------------------+
+    |9a3b642c-06c6-11ee-be56-0242ac120002|
+    +------------------------------------+
+    ```
+
+## mask_canada_sin(str [,mask_char])
+
+
+
+Masks a Canada Social Insurance Number (SIN).
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | The masking character | String |
+
+The `str` accepts an alphanumeric string.
+
+If the `mask_char` is not specified, the default is `X`.
+
+### Returns
+
+A string with the selected characters masked by a specified `mask_char` or that option's default value.
+
+Returns an error if the length of `str` is not correct.
+
+Returns a NULL value if the `str` is in an incorrect length. If the `str` contains a multi-byte character, the `str` is not masked.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT mask_canada_sin('555-555-555');
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +--------------------------------+
+    | mask_canada_sin('555-555-555') |
+    +--------------------------------+
+    | XXX-XXX-XXX                    |
+    +--------------------------------+
+    ```
+
+## mask_iban(str [,mask_char])
+
+
+
+Masks an Internal Bank Account Number (IBAN).
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | Character used for masking | String |
+
+The `str` accepts either of the following:
+
+* No separator symbol
+* Groups of four characters. These groups can be separated by a space or any separator character.
+
+A `str` 
+The default value for `mask_char` is `*`.
+
```

Comment:
mask_char can be a multibyte character in any character set (not necessarily identical to the character set of str)